File: revision-slides/revision-week-1/solutions/average.py
a = input("First number: ")
b = input("Second number: ")

a = int(a)                      # input() always gives a str
b = int(b)                      # forget these and a + b is "72", with no error

total = a + b
average = total / 2             # Python's / gives 4.5 here; Java's gave 4

print("Sum: " + str(total))     # without str(): TypeError, str + int
print("Average: " + str(average))

File: revision-slides/revision-week-1/solutions/Average.java
import java.util.Scanner;

public class Average {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        System.out.print("First number: ");
        int a = input.nextInt();

        System.out.print("Second number: ");
        int b = input.nextInt();

        int sum = a + b;

        double average = sum / 2.0;     // or: (double) sum / 2
        // sum / 2 is int / int, so 9 / 2 gives 4 -- the Week 1 trap

        System.out.println("Sum: " + sum);
        System.out.println("Average: " + average);
    }
}
